tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -531,4 +531,48 @@ public function data_provider_not_supported_webp_images() {
 		yield 'PNG image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png' );
 		yield 'GIFT image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/earth.gif' );
 	}
+
+	/**
+	 * Checks whether the sources information is added to image sizes details of the REST response object.
+	 *
+	 * @test
+	 */
+	public function it_should_add_sources_to_rest_response() {
+		$file_location = TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg';
+		$attachment_id = $this->factory->attachment->create_upload_object( $file_location );
+		$metadata      = wp_get_attachment_metadata( $attachment_id );
+
+		$request       = new WP_REST_Request();
+		$request['id'] = $attachment_id;
```

Comment:
Personally due `$request` is an object I would go with the object approach:

```suggestion
		$request->set_param( 'id', $attachment_id );
```

Reply:
The `WP_REST_Request` implements the `ArrayAccess` interface and `$request[...] = ...` assignments are valid as well as direct `set_param` method calls. REST controllers in the core also use array access approach, see here: https://github.com/WordPress/wordpress-develop/blob/5.9/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L555

Reply:
Good point. :heavy_check_mark: